src/Components/Reports/Util.js

```diff
@@ -70,3 +70,68 @@ export const getSystemsReportFileName = () =>  {
 
     return reportFileName;
 };
+
+export const formatExecutiveReportData = (data) => {
+
+    const { systems_per_state: systemsPerState, conditions } = data;
+
+    const stateChartData = [];
+    const stateTableData = [['# of systems']];
+
+    const conditionsChartData = [];
+    const conditionsTableData = [['# of occurrences']];
+
+    const condtionsInfo = {
+        io: {
+            name: 'Disk IO',
+            occurances: []
+        },
+        memory: {
+            name: 'RAM',
+            occurances: []
+        },
+        cpu: {
+            name: 'CPU',
+            occurances: []
+        }
+    };
+
+    const stateNameMapping = {
+        idling: 'Idling',
+        oversized: 'Oversized',
+        under_pressure: 'Under pressure', /* eslint-disable-line camelcase */
+        undersized: 'Undersized',
+        waiting_for_data: 'Waiting for data', /* eslint-disable-line camelcase */
+        optimized: 'Optimized'
+    };
+
+    Object.keys(systemsPerState).map((state) => {
+        const stateName = stateNameMapping[state];
+        const percentage = Math.floor(systemsPerState[state]?.percentage);
+        const count = systemsPerState[state]?.count;
+
+        stateChartData.push({ x: stateName, y: percentage });
+        stateTableData.push([`${count} (${percentage}% of total)`]);
+    });
+
+    Object.keys(conditions).map((condition) => {
+        const conditionName = condtionsInfo[condition].name;
+        const percentage = Math.floor(conditions[condition]?.percentage);
+        const count = conditions[condition]?.count;
+
+        conditionsChartData.push({ x: conditionName, y: percentage });
+        conditionsTableData.push([`${count}`]);
+
+        const underPressureValue = conditions[condition].under_pressure ? conditions[condition].under_pressure : 0;
+        const undersizedValue = conditions[condition].undersized ? conditions[condition].undersized : 0;
+        const oversizedValue = conditions[condition].oversized ? conditions[condition].oversized : 0;
+
+        condtionsInfo[condition].occurances.push(['Under pressure', `${underPressureValue}`]);
+        if (condition !== 'io') {
```

Comment:
```suggestion
        if (condition === 'io' && undersizedValue != -1 && oversizedValue != -1) {
```

Using this condition, I think we won't need to touch frontend once backend is ready for IOPS. What do you think?

Reply:
good point @kgaikwad I have made the above change and will push it in the latest commit 💯